egui/src/widgets/plot/mod.rs

```diff
@@ -100,31 +99,50 @@ impl Plot {
         }
     }
 
-    fn auto_color(&mut self, color: &mut Color32) {
-        if *color == Color32::TRANSPARENT {
-            let i = self.next_auto_color_idx;
-            self.next_auto_color_idx += 1;
-            let golden_ratio = (5.0_f32.sqrt() - 1.0) / 2.0; // 0.61803398875
-            let h = i as f32 * golden_ratio;
-            *color = Hsva::new(h, 0.85, 0.5, 1.0).into(); // TODO: OkLab or some other perspective color space
-        }
+    fn auto_color(&mut self) -> Color32 {
+        let i = self.next_auto_color_idx;
+        self.next_auto_color_idx += 1;
+        let golden_ratio = (5.0_f32.sqrt() - 1.0) / 2.0; // 0.61803398875
+        let h = i as f32 * golden_ratio;
+        Hsva::new(h, 0.85, 0.5, 1.0).into() // TODO: OkLab or some other perspective color space
     }
 
     /// Add a data curve.
     /// You can add multiple curves.
     pub fn curve(mut self, mut curve: Curve) -> Self {
-        if !curve.no_data() {
-            self.auto_color(&mut curve.stroke.color);
-            self.curves.push(curve);
+        if curve.no_data() {
+            return self;
+        };
+
+        // Give the stroke an automatic color if no color has been assigned.
+        let stroke_color = curve.color.get_or_insert_with(|| self.auto_color());
+        // Give the marker the same color as the stroke if no color has been assigned.
+        if let Some(marker) = &mut curve.marker {
+            if marker.color == Color32::TRANSPARENT {
+                marker.color = if *stroke_color != Color32::TRANSPARENT {
+                    *stroke_color
+                } else {
+                    self.auto_color()
+                };
+            }
         }
+        self.curves.push(curve);
+
         self
     }
 
+    /// Add multiple data curves.
+    pub fn curves(self, curves: Vec<Curve>) -> Self {
+        curves.into_iter().fold(self, Self::curve)
+    }
+
     /// Add a horizontal line.
     /// Can be useful e.g. to show min/max bounds or similar.
     /// Always fills the full width of the plot.
     pub fn hline(mut self, mut hline: HLine) -> Self {
-        self.auto_color(&mut hline.stroke.color);
+        if hline.stroke.color.a() == 0 {
+            hline.stroke.color = self.auto_color();
```

Comment:
this will overwrite additive colors

Reply:
Do you mean I shouldn't check if alpha == 0 and instead compare against `Color32::TRANSPARENT`?

Reply:
yeah, exactly - `hline.stroke.color.a() == 0` is true for transparent colors, but also for additive colors (which is pretty nice to use in a plot with dark background). `color == Color32::TRANSPARENT` is the correct check here